### DISCUSSION TITLE
My NanoEls-H2 build

### DISCUSSION THREAD

**Discussion by EdFleta** [kachurovskiy/nanoels#87] [Show and tell]
Hi everybody,
Here are some pictures of how I installed an electronic lead screw on my MX210V from Vevor.
As said before, I took advantage of the fact that this particular version has a aluminum hollow frame. The spindle encoder could be tucked into the hollow space and maintained in place thanks to a bespoke 3d printed part (PETG).

The stepper is a Cloudray Nema 23 (closed loop) with the recommended driver. I run it at 800 steps (the driver does not allow 200) and I wonder whether it could be of interest to use higher stepping modes. to be tested.

The rest is straightforward with all the components and wiring enclosed in the original control cabinet.

![Front](https://user-images.githubusercontent.com/113701009/202857280-dc67f1d8-1c09-4799-887a-55f6010eed5f.png)

![Back](https://user-images.githubusercontent.com/113701009/202857669-ef6f8a2a-79cd-431a-9588-fae8bf6fd6eb.png)
![Inner](https://user-images.githubusercontent.com/113701009/202857678-3c0a687a-b81d-40cf-988c-9ec6b1d6d527.png)
![stepper](https://user-images.githubusercontent.com/113701009/202857688-2f0f5ca3-48c5-4051-ad3d-30ca7eb09ac3.png)
![box](https://user-images.githubusercontent.com/113701009/202857693-8f156205-146b-4e42-8715-2899b38b08c7.png)

**Comment by kachurovskiy** [comment]
Wow, this could be the tidiest NanoEls build I've even seen... Well done!

**Comment by Dididu12** [comment]
Bravo 👏 cela et très propre !!!

**Comment by fmw626** [comment]
Very nice build Would you mind sharing the encoder mount model? I am currently also making a build on the mx-210v. Thank you!

**Reply by EdFleta** [reply]
With pleasure. Please find the file attached
[chassis_plug_VF.zip](https://github.com/kachurovskiy/nanoels/files/10483286/chassis_plug_VF.zip)

**Reply by fmw626** [reply]
Awesome, thank you very much!

**Comment by fraidd** [comment]
bonjour
ou avez vous trouvé les stl des poulies ? encodeur et broche?
merci a vous
fraidd

**Reply by EdFleta** [reply]
The encoder and spindle pulleys are bespoke made using a generic pulley designer tool you can find on Thingverse. I then slightly modified them with Freecad and inserted an aluminum hub in the encoder one. I can send you the stl but they may be useless if you do not use the same gear ratio.

**Reply by fraidd** [reply]
ok merci

**Reply by fraidd** [reply]
ou alors le lien thingverse ..

**Reply by EdFleta** [reply]
[Pulley_T-MXL-XL-HTD-GT2_80-tooth.zip](https://github.com/kachurovskiy/nanoels/files/11004580/Pulley_T-MXL-XL-HTD-GT2_80-tooth.zip)

Voici l'outil en question. Il faudra bien entendu adapter les paramètres. s'ouvre avec OPENSCAD. Good luck !

**Comment by fraidd** [comment]
merci a vous je vais tester et ça ne peut qu'améliorer mes connaissances
fraidd

**Reply by Salemlt** [reply]
Would you be willing to share the stl for the spindle gear? What belt are you using for the encoder and lead screw?
I have the same lathe running the encoder from the change gear, but I? love your setup so clean and slick unlike mine lol

Thank you in advance

**Comment by Salemlt** [comment]
Would you be willing to share the stl for the spindle gear? What belt are you using for the encoder and lead screw?
I have the same lathe running the encoder from the change gear, but I? love your setup so clean and slick unlike mine lol

Thank you in advance

**Reply by EdFleta** [reply]
Hi, I have already given all the infos in the thread above your message. (but if you have the very same lathe then you can use my stl in the zip attached)
In addition , the encoder belt is a 2GT-340mm , 6mm width whilst the lead screw one is a 5M-180-15mm
[Pulley_GT2_spindle_encoder.zip](https://github.com/kachurovskiy/nanoels/files/11927240/Pulley_GT2_spindle_encoder.zip)
best regards, eddie

**Reply by Salemlt** [reply]
Thank you!
What material did you use to print with?

**Reply by EdFleta** [reply]
I have used PLA although  everybody would recommend using PETG. but no problem so far

**Reply by X3msnake** [reply]
@EdFleta and there should be no problems in the future since there are no torque/strain associated with the motion unless you have the machine in a hot place (above 60ºc) then PLA can deform, lovely assembly by the way and great little lathe, i have one like it :)

https://photos.app.goo.gl/Nx5b19CaSwa68SFE7

Need to make the same mod as you did someday soon

**Reply by Salemlt** [reply]
Thank you so much!
Great work, you have an eye for detail!

**Comment by gregreid** [comment]
I just happened across this page (while Google searching to try and find more detail on my MX-210V spindle).  Beautiful job.

I too added an ELS to my MX-210V, and it turns out that we had very similar solutions.  I did mine in May 2022, so I may have beat you by a few months, but they were clearly done independently.  I didn't own a milling machine at the time so my motor mount isn't nearly as nice as yours.  

My encoder is mounted in the same convenient casting cavity as yours, but used gear drive to it rather than belt drive, with a couple of 3D printed gears to make up the 1:1 drive train along with two out-of-box metal gears.  

I used a servo motor (the model number is visible in one of the pictures) with a 10T to 60T reduction belt drive.  (Sorry I didn't bother removing that cover to show the 10T pulley on the motor shaft).

My ELS is from Clough42 on YouTube, which was simple to implement.  The Texas Instruments control board lives inside the rear cabinet, same as yours (though I mounted mine on an inside surface rather than the back panel as yours.  I installed a small fan in the side of the rear cabinet ant cut vent slits in the back to help cool things in there.  But again, very similar ideas and executions.  

I'll attach a few pics of mine just FYI and FWIW.  Sorry that the lathe is VERY DIRTY at the moment. :-/

![IMG_6264](https://github.com/kachurovskiy/nanoels/assets/9823373/94f5cc8e-cb68-426d-9da2-fc484057c983)
![IMG_6261](https://github.com/kachurovskiy/nanoels/assets/9823373/da988f2f-fc36-4d3f-aef9-255077b8430f)
![IMG_6263](https://github.com/kachurovskiy/nanoels/assets/9823373/6a94ec2a-583b-485e-875b-dabd2ef4789c)
![IMG_6262](https://github.com/kachurovskiy/nanoels/assets/9823373/5ce73e7d-2e3d-4cf4-b7ad-2ff12e66405c)


Oh, and from an October 2022 posting that I'd made to the Hobby Machinist group about my ELS setup, I remember now that I'd also recorded some videos of my setup.  Here's the quote from Hobby Machinist with the link and detail, FWIW:

--------------

Sorry it took so long, but I finally got around to making some videos of my lathe DRO and ELS (and etc.) setup. Here's a link that should bring you to SIX separate videos: https://photos.app.goo.gl/wqaXNRgXaXCiFEot7 

The first two are for the DRO: it seems that I accidentally hit the STOP button on my iPhone during the first recording as I needed three hands to do the measuring while holding the phone, so the second short video is really an extension of the first. Then the third video describes the ELS setup, extended a bit into an important extra note at the start of the 4th video. The fifth and sixth vids are just "fluff" that you can skip.

**Reply by EdFleta** [reply]
Hi Greg, 
Nice job indeed. I too considered using the Clough42 solution but I found the BOM much more expensive than Max Kachurovskiy's one. In addition the parts and the TI board in particular are quite difficult to find in Europe.
As for the stepper mount no milling machine is required. I started from a square piece of 10mm thick aluminum and shaped it with basic tooling.
best regards,

eddie

**Reply by gregreid** [reply]
Ya, the TI board was hard to get at the time (COVID shipping issues).  I visited the ti.com website multiple times every day to be ready when they came back into stock.  I ordered another from digikey and got on their back-order list.  I was finally able to snap one up on ti.com directly, and the digikey order was fulfilled some months later.  So I now have a spare "just in case".

I should've spent more time on my servo mount.  I just wanted to get it working so I could cut some oddball threads that I needed at the time for an airplane landing gear project.  So it's a trivial welded aluminum bracket screwed down onto the benchtop.  There's no belt tensioner adjustment, but it just "works".  ;-)

**Comment by Dididu12** [comment]
Hi Maxim I wanted to know if it was possible to put a nema 23 in place of a
nema 17 on the x axis what should be modified to the code mega els. I would
like to put it live thank you

**Reply by kachurovskiy** [reply]
Hi Dilan, yes, it should be possible, just make sure the driver can receive the same signals and can be connected in the same way (check driver datasheet).

**Comment by Dididu12** [comment]
hello maxim, and can it be possible with mega els to add threads of more
than 4mm to make metal forests it's to make the helical form of the forest
and 4mm of footsteps is not enough it takes more

Le dim. 3 nov. 2024, 08:43, Maxim Kachurovskiy ***@***.***> a
écrit :

**Reply by kachurovskiy** [reply]
I have no idea how to change this in the code but you can simply decrease your stepper driver step count 2x to get a 2x larger thread pitch.

**Comment by Dididu12** [comment]
Ok merci et combien coûterait a faire moi même la version h4 ?

Le ven. 8 nov. 2024, 08:22, Maxim Kachurovskiy ***@***.***> a
écrit :

**Reply by kachurovskiy** [reply]
It depends, if you are up for the challenge you can get a ESP32-S3 dev board from AliExpress for 5 EUR and Adafruit TCA8418 for 15 EUR. Display another 5 EUR. And a lot of wiring for the 39 buttons 😄

**Comment by Dididu12** [comment]
Oui intéressant

Le ven. 8 nov. 2024, 08:57, Maxim Kachurovskiy ***@***.***> a
écrit :

**Comment by Dididu12** [comment]
there is nothing to modify at the level of nema 17 and 23 compare to
megaels?

Le ven. 8 nov. 2024, 12:36, Dilan Delabrousse ***@***.***> a
écrit :
